packages/contracts/galaxy-oracle/src/lib.rs

```
@@ -0,0 +1,345 @@
+
+        buffer.push(
+            PriceEntry {
+                price,
+                timestamp: env.ledger().timestamp(),
+                pusher: pusher.clone(),
+            },
+            TWAP_WINDOW_SIZE,
+        );
+
+        prices.set(key, buffer);
+        storage.set(&KEY_PRICES, &prices);
```

Clippy (1.96.0)

[warning] manual `!RangeInclusive::contains` implementation

(warning)